dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/DatadogCoreProxy.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core
+
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.storage.EventBatchWriter
+import com.datadog.android.api.storage.RawBatchEvent
+import com.google.gson.Gson
+import com.google.gson.JsonElement
+
+class DatadogCoreProxy(
+    val core: InternalSdkCore,
+) : InternalSdkCore by core {
+    private val featureScopes = mutableMapOf<String, FeatureScopeInterceptor>()
+
+    fun eventsWritten(featureName: String): String {
+        val events = featureScopes[featureName]?.eventsWritten()?.toList() ?: emptyList<Any>()
```

Comment:
It is better to have `toList` call in the `FeatureScopeInterceptor#eventsWritten` instead, so it can provide immutable copy by design. Also there is handy `orEmpty` method.

```suggestion
        val events = featureScopes[featureName]?.eventsWritten().orEmpty()
```